juriscraper/opinions/united_states/state/colo.py

```diff
@@ -23,25 +26,61 @@ def __init__(self, *args, **kwargs):
-
         date = self.html.xpath(
             ".//p[contains(text(), 'case announcements for')]/text()"
-        )[0]
-        date_str = (
-            re.findall(r"case announcements for (.*) are now", date)[0]
-            .split(",", 1)[1]
-            .strip()
         )
-        for item in self.html.xpath(
-            ".//a[contains(@href, 'Supreme_Court/Opinions/')]"
-        ):
-            docket, name = item.text_content().split(" ", 1)
-            self.cases.append(
-                {
-                    "date": date_str,
-                    "docket": docket,
-                    "name": name,
-                    "url": item.attrib["href"],
-                    "status": "Published",
-                }
+        if date:
+            date_str = (
+                re.findall(r"case announcements for (.*) are now", date[0])[0]
+                .split(",", 1)[1]
+                .strip()
             )
+            for item in self.html.xpath(
+                ".//a[contains(@href, 'Supreme_Court/Opinions/')]"
+            ):
+                link_text = normalize_dashes(item.text_content().strip())
+                cite_match = re.findall(r"(\d{2,4}\s*CO\s*\d+\w?)", link_text)
+                docket_match = re.findall(r"\d+[A-Z]+\d{2,}", link_text)
+                docket = docket_match[0] if docket_match else ""
+                citation = cite_match[0] if cite_match else ""
+                name = (
+                    link_text.replace(docket, "")
+                    .replace(citation, "")
+                    .lstrip(",- ")
+                )
+                self.cases.append(
+                    {
+                        "date": date_str,
+                        "docket": docket,
+                        "name": name,
+                        "url": item.attrib["href"],
+                        "status": "Published",
+                        "citation": citation,
+                    }
+                )
+
+    def extract_from_text(self, scraped_text: str) -> Dict[str, Any]:
+        """Pass scraped text into function and return data as a dictionary
+        Notes for 'Citation':
+            - Reporter key for this court: 'CO'
+            - Type for a state: 2
+        :param scraped_text: Text of scraped content
+        :return: metadata
+        """
+        cite_match = re.findall(r"\d{4}\s*CO\s*\d+", scraped_text)
+        docket_match = re.findall(r"\d+S[A-Z]\d+", scraped_text)
+        metadata = {}
+        if cite_match and docket_match:
+            volume, page = cite_match[0].split("CO")
+            metadata = {
+                "OpinionCluster": {
+                    "docket_number": docket_match[0],
+                },
+                "Citation": {
+                    "volume": volume.strip(),
+                    "reporter": "CO",
+                    "page": page.strip(),
+                    "type": 2,
+                },
+            }
+        return metadata
```

Comment:
this is better, but what happens if the opinion doesn't have a docket number? or we get a permutation that we didn't expect?  

I think we should process both regexes separately to make the metadata object.

Reply:
Removed search for the docket.